modules/ROOT/pages/subqueries/subqueries-in-transactions.adoc

```diff
@@ -591,6 +602,189 @@ RETURN n.num, s.errorMessage;
+
+[source, cypher]
+----
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS row
+CALL {
+  WITH row
+  CREATE (p:Person {tmdbId: row.person_tmdbId})
+  SET p.name = row.name, p.born = row.born
+} IN 3 CONCURRENT TRANSACTIONS OF 10 ROWS
+RETURN count(*) AS personNodes
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="m"]
+|===
+| personNodes
+| 444
+1+d| Rows: 1
+|===
+
+====
+
+[[non-determinism]]
+=== Concurrency and non-deterministic results
+
+`CALL { ... } IN TRANSACTIONS` uses _ordered_ semantics by default, where batches are committed in a sequential row-by-row order.
+For example, in `CALL { <I> } IN TRANSACTIONS`, any writes done in the execution of `<I~1~>` must be observed by `<I~2~>`, and so on.
+
+In contrast, `CALL { ... } IN CONCURRENT TRANSACTIONS` uses _concurrent_ semantics, where both the number of rows committed by a particular batch and the order of committed batches is undefined.
+That is,  in `CALL { <I> } IN CONCURRENT TRANSACTIONS`, writes committed in  execution of `<I~1~>` may not be observed by `<I~2~>`, and so on.
+
+The results of `CALL` subqueries executed in concurrent transactions may, therefore, not be deterministic.
+To guarantee deterministic results, ensure that the results of committed batches are not dependent on each other.
+
+[NOTE]
+Using `CALL { ... } IN CONCURRENT TRANSACTIONS` can impact xref:subqueries/subqueries-in-transactions.adoc#error-behavior[error behavior.]
+Specifically, when using `ON ERROR BREAK` or `ON ERROR FAIL` and one transaction fails, then any concurrent transactions _may_ not be interrupted and rolled back (though subsequent ones would).
+This is because no timing guarantees can be given for concurrent transactions.
+That is, an ongoing transaction may or may not commit successfully in the time window when the error is being handled.
+Use the xref:subqueries/subqueries-in-transactions.adoc#status-report[status report] to determine which batches were committed and which failed or did not start.
+
+
+[[deadlocks]]
+=== Deadlocks
+
+When a write transaction occurs, Neo4j takes locks to preserve data consistency while updating.
+For example, when creating or deleting a relationship, a write lock is taken on both the specific relationship and its connected nodes.
+
+A deadlock happens when two transactions are blocked by each other because they are attempting to concurrently modify a node or a relationship that is locked by the other transaction (for more information about locks and deadlocks in Neo4j, see link:{neo4j-docs-base-uri}/operations-manual/{page-version}/database-internals/concurrent-data-access/#_locks[Operations Manual -> Concurrent data access].
+
+A deadlock may occur when using `CALL { ... } IN CONCURRENT TRANSACTIONS` if the transactions for two or more batches tries to take the same lock on a node or relationship.
```

Comment:
I think "to take the same locks in an order that results in a circular dependency between them" is more correct, 
as if they just try to take the same lock one could wait for the other and both would succeed.